File: codility/tapeequ.py
#!/usr/bin/env python

def solution(A):
    #print A
    N = len(A)
    S = []
    sum = 0
    for i in xrange(N):
        S.append(sum + A[i])
        sum += A[i]
    #print S
    mm = 2147483647
    for i in xrange(1, N):
        left = S[i - 1]
        right = S[N - 1] - S[i - 1]
        diff = abs(left - right)
        mm = min(mm, diff)
    return mm

if __name__ == '__main__':
    print solution([3, 1]) == 2
    print solution([3, 1, 2, 4, 3]) == 1

File: codility/tapeequ.cc
#include <algorithm>
#include <iostream>
#include <map>
#include <vector>

using namespace std;

int solution(vector<int> &A) {
    int N = (int) A.size();
    vector<int> sum;
    int i;
    int s = 0;
    for(i = 0; i < N; i++) {
        s += A[i];
        sum.push_back(s);
    }
    int mn = 20000;
    for(i = 1; i < N; i++) {
        int d = abs(sum[i-1] - (sum[N-1] - sum[i-1]));
        mn = min(d, mn);
    }
    return mn;
}
